app/scripts/modules/core/cluster/task.matcher.ts

```diff
@@ -0,0 +1,92 @@
+import {has, intersection, map} from 'lodash';
+
+import {ITask} from '../task/task.read.service';
+import {ServerGroup} from '../domain/serverGroup';
+
+export interface ITaskMatcher {
+ (task: ITask, serverGroup: ServerGroup): boolean;
+}
+
+function createcopylastasgMatcher(task: ITask, serverGroup: ServerGroup): boolean {
+  const source = task.getValueFor('source'),
```

Comment:
The compiler/linter doesn't complain because `.getValueFor()` returns `any`. I'll add an `any` here for consistency.